nixos/modules/services/misc/huginn.nix

```diff
@@ -0,0 +1,728 @@
+	if [ "${cfg.appSecretToken}" = "" ]; then
+	  printf "APP_SECRET_TOKEN=" >>  ${cfg.statePath}/tmp/secrets
+          tr -dc A-Za-z0-9 < /dev/urandom 2> /dev/null | head -c 32 >> ${cfg.statePath}/tmp/secrets
+	  echo "" >> ${cfg.statePath}/tmp/secrets
+	fi
+        if [ "${cfg.databasePassword}" = "" ]; then
+	  printf "DATABASE_PASSWORD=" >> ${cfg.statePath}/tmp/secrets
+	  if ! test -e "${cfg.statePath}/db-created"; then
+	    tr -dc A-Za-z0-9 < /dev/urandom 2> /dev/null | head -c 32 > ${cfg.statePath}/db-password
+	  fi
+          cat ${cfg.statePath}/db-password >> ${cfg.statePath}/tmp/secrets
+	  echo "" >> ${cfg.statePath}/tmp/secrets
+        fi
+        if ! test -e "${cfg.statePath}/invitation_code"; then
+           if [ "${cfg.registration.invitationCode}" = "" ]; then
+             tr -dc A-Za-z0-9 < /dev/urandom 2> /dev/null | head -c 32 > ${cfg.statePath}/invitation_code
+	   fi
+	fi
+	if [ "${cfg.registration.invitationCode}" = "" ]; then
+	  printf "INVITATION_CODE=" >> ${cfg.statePath}/tmp/secrets
+          cat ${cfg.statePath}/invitation_code >> ${cfg.statePath}/tmp/secrets
+	  echo "" >> ${cfg.statePath}/tmp/secrets
+        fi
+
+	 if [ "${cfg.databaseHost}" = "127.0.0.1" ]; then
+          if ! test -e "${cfg.statePath}/db-created"; then
+	    if [ "${cfg.databasePassword}" = "" ]; then
+	      psql postgres -c "CREATE ROLE huginn WITH LOGIN CREATEDB NOCREATEROLE NOCREATEUSER ENCRYPTED PASSWORD '`cat ${cfg.statePath}/db-password`'"
+	    else
+              psql postgres -c "CREATE ROLE huginn WITH LOGIN CREATEDB NOCREATEROLE NOCREATEUSER ENCRYPTED PASSWORD '${cfg.databasePassword}'"
+	    fi
+            ${config.services.postgresql.package}/bin/createdb --owner ${cfg.databaseUser} ${cfg.databaseName} huginn  || true
+	    if [ "${cfg.databasePassword}" = "" ]; then
+	      echo "creating db"
+              ${huginn-rake}/bin/huginn-rake db:create RAILS_ENV=production APP_SECRET_TOKEN="null" DATABASE_PASSWORD=`cat ${cfg.statePath}/db-password`
+	      echo "migrating db"
+	      ${huginn-rake}/bin/huginn-rake db:migrate RAILS_ENV=production APP_SECRET_TOKEN="null" DATABASE_PASSWORD=`cat ${cfg.statePath}/db-password`
+	      echo "seeding db"
+	      ${huginn-rake}/bin/huginn-rake db:seed RAILS_ENV=production APP_SECRET_TOKEN="null" SEED_USERNAME=${cfg.initialUser} SEED_PASSWORD=${cfg.initialPassword} DATABASE_PASSWORD=`cat ${cfg.statePath}/db-password` SKIP_INVITATION_CODE=true REQUIRE_CONFIRMED_EMAIL=false
+	    else
+	      echo "creating db"
+              ${huginn-rake}/bin/huginn-rake db:create RAILS_ENV=production APP_SECRET_TOKEN="null"
+	      echo "migrating db"
+	      ${huginn-rake}/bin/huginn-rake db:migrate RAILS_ENV=production APP_SECRET_TOKEN="null"
+	      echo "seeding db"
+	      ${huginn-rake}/bin/huginn-rake db:seed RAILS_ENV=production APP_SECRET_TOKEN="null" SEED_USERNAME=${cfg.initialUser} SEED_PASSWORD=${cfg.initialPassword} SKIP_INVITATION_CODE=true REQUIRE_CONFIRMED_EMAIL=false
```

Comment:
what happens if you pass `DATABASE_PASSWORD=` to rake? If this works you could save the if expression.